File: Send Email/[Objective-C] sendEmail/sendEmail/ViewController.m
//
//  ViewController.m
//  sendEmail
//
//  Created by Luis Andres Castillo Hernandez on 10/29/15.
//  Copyright © 2015 DevBros. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

@synthesize messageLabel;
@synthesize imageSelected;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


-(void)viewDidAppear:(BOOL)animated
{

    imageSelected.image = [UIImage imageNamed:@"StarWars.jpg"];

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}





/* sending email with NO attachment */
- (IBAction)sendEmail:(UIButton *)sender
{

    MFMailComposeViewController *composer = [[MFMailComposeViewController alloc]init];
    
    [composer setMailComposeDelegate:self];
    
    //checking if device is avaliable for sending email
    
    if(  [MFMailComposeViewController canSendMail] )
    {
        
        NSArray * receipents = [[NSArray alloc]initWithObjects:@" ", nil];
        
        [composer setToRecipients:receipents];
        [composer setSubject:@"This is a test email"];
        [composer setMessageBody:messageLabel.text isHTML:NO];
        
        //
        [composer setModalTransitionStyle:UIModalTransitionStyleCrossDissolve];
        [self presentViewController:composer animated:YES completion:NULL];
    }
    else
    {
        NSLog(@"This Device can NOT send emails!");
    }
    
}//eo-action


/* sending email with image attachment */
- (IBAction)sendEmailWithImage:(UIButton *)sender
{

    MFMailComposeViewController *composer = [[MFMailComposeViewController alloc]init];
    
    [composer setMailComposeDelegate:self];
    
    //checking if device is avaliable for sending email
    
    if(  [MFMailComposeViewController canSendMail] )
    {
        
        NSArray * receipents = [[NSArray alloc]initWithObjects:@" ", nil];
        
        [composer setToRecipients:receipents];
        [composer setSubject:@"This is a test email"];
        [composer setMessageBody:messageLabel.text isHTML:NO];
        
        
        //preparing image
        NSData * imageData = UIImageJPEGRepresentation(self.imageSelected.image, 1);
        
        //adding image
        [composer addAttachmentData:imageData mimeType:@"image/jpg" fileName:@"imageOfStarWars.jpg"];
        
        //
        [composer setModalTransitionStyle:UIModalTransitionStyleCrossDissolve];
        [self presentViewController:composer animated:YES completion:NULL];
    }
    else
    {
        NSLog(@"This Device can NOT send emails!");
    }

}//eo-action





//MARK: mail compose delegates
-(void)mailComposeController:(MFMailComposeViewController *)controller didFinishWithResult:(MFMailComposeResult)result error:(NSError *)error
{

    [self dismissViewControllerAnimated:YES completion:NULL];

    //error
    if(error)
    {
        NSLog(@"%@", [error description]);
    }
    
    
    
}//eom

@end

File: Send Text Message/[Swift] sendSMS/sendSMS/ViewController.swift
//
//  ViewController.swift
//  sendSMS
//
//  Created by Luis Castillo on 1/16/16.
//  Copyright © 2016 DevBros. All rights reserved.
//

import UIKit
import MessageUI

class ViewController: UIViewController, MFMessageComposeViewControllerDelegate
{
    
    
    //MARK: - View Loading
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
    }//eom
    
    override func viewDidAppear(animated: Bool) {
        
    }//eom
    
    
    //MARK: -
    func sendSMS()
    {
        let smsComposer:MFMessageComposeViewController = MFMessageComposeViewController()
    
        smsComposer.messageComposeDelegate = self
        
        if MFMessageComposeViewController .canSendText()
        {
            let recipients:[String] = ["111-222-3321","391-221-2091"]
            
            //recipients
            smsComposer.recipients = recipients
            
            //subject
            smsComposer.subject = "subject goes here"
            
            //body
            smsComposer.body = "Hey there it's me!"
            
            //attachments
            //smsComposer.addAttachmentData(<#T##attachmentData: NSData##NSData#>, typeIdentifier: <#T##String#>, filename: <#T##String#>)
            
            //presenting message
            self .presentViewController(smsComposer, animated: true, completion: { () -> Void in
                
            })
        }
        else
        {
            print("????")
        }
    }//eom

    
    
    //MARK: - SMS Delegates
    func messageComposeViewController(controller: MFMessageComposeViewController, didFinishWithResult result: MessageComposeResult)
    {
        switch result
        {
        case MessageComposeResultCancelled:
            print("message Cancelled")
            break
        case MessageComposeResultFailed:
            print("message failed")
            break
        case MessageComposeResultSent:
            print("message sent")
            break
        default :
            print("unknown message occurred")
            break
        }
        
        self.dismissViewControllerAnimated(true) { () -> Void in
            
        }
        
    }//eom
    
    
    @IBAction func sendMessage(sender: AnyObject)
    {
        self.sendSMS()
    }//eo-a
    
    
    
    //MARK: - Memory
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

